AsyncNats/Util/NatsPublishChannel.cs

```diff
@@ -0,0 +1,122 @@
+﻿namespace EightyDecibel.AsyncNats
+{
+    using EightyDecibel.AsyncNats.Messages;
+    using System;
+    using System.Collections.Concurrent;
+    using System.Threading;
+    using System.Threading.Channels;
+    using System.Threading.Tasks;
+
+
+    public delegate void SerializeDelegate(Span<byte> writeBuffer);
+
+    internal class NatsPublishChannel
+    {
+        public int DefaultBufferLength { get; set; } = 1024 * 64;
+
+        readonly struct NatsPublishAction
+        {
+            public readonly int Length;
+            public readonly SerializeDelegate Delegate;
+
+            public NatsPublishAction(int length, SerializeDelegate del)
+            {
+                Length = length;
+                Delegate = del;
+            }
+        }
+
+        public ChannelReader<NatsPublishBuffer> Reader => _channel.Reader;
+        public ChannelWriter<NatsPublishBuffer> Writer => _channel.Writer;
+
+
+        readonly Channel<NatsPublishBuffer> _channel;
```

Comment:
Missing visibility keyword [private]

Reply:
Only mentioned it here but it's in several classes, explicit declaration of visibility is preferred

Reply:
Right, ran code cleanup adding access modifier to all fields.